File: akshar-app/packages/mesh/src/middleware/error-handler.ts
/**
 * Global Express error handler — fail-closed, generic responses.
 */
import { Request, Response, NextFunction } from 'express';

export function errorHandler(err: Error, _req: Request, res: Response, _next: NextFunction): void {
  console.error('[ERROR]', err.message);
  res.status(500).json({ ok: false, error: 'Internal server error' });
}

File: akshar-app/packages/auth/app/middleware/error_handler.py
"""Global exception handler — fail-closed, generic error responses."""
from __future__ import annotations

import logging
import traceback

from fastapi import Request
from fastapi.responses import JSONResponse
from starlette.middleware.base import BaseHTTPMiddleware

logger = logging.getLogger("akshar.auth")


class ErrorHandlerMiddleware(BaseHTTPMiddleware):
    """Catches all unhandled exceptions and returns generic 500 responses."""

    async def dispatch(self, request: Request, call_next):
        try:
            return await call_next(request)
        except Exception as exc:
            # Log full traceback internally
            logger.error(
                "Unhandled exception",
                extra={
                    "path": request.url.path,
                    "method": request.method,
                    "error": str(exc),
                    "traceback": traceback.format_exc(),
                },
            )
            # Return generic error (SECURITY-09: no stack traces to client)
            return JSONResponse(
                status_code=500,
                content={"ok": False, "error": "Internal server error"},
            )
